src/repository/opamDownload.ml

```diff
@@ -193,3 +193,169 @@ let download ?quiet ?validate ~overwrite ?compress ?checksum url dstdir =
   in
   download_as ?quiet ?validate ~overwrite ?compress ?checksum url dst @@|
   fun () -> dst
+
+
+(** Stdout output retrieval and post requests management *)
+
+let post_tools = ["wget"; "curl"]
+let check_post_tool () =
+  match Lazy.force OpamRepositoryConfig.(!r.download_tool) with
+  | [(CIdent cmd), _], _ -> List.mem cmd post_tools
+  | _ -> false
+
+let get_output ~post ?(args=[]) url =
+  let cmd_args =
+    download_args ~url ~out:"-" ~retry:OpamRepositoryConfig.(!r.retries)
+      ~compress:false ()
+    @ args
+  in
+  let cmd_args =
+    if post then
+      match cmd_args with
+      | ("wget"|"curl" as cmd)::args -> Some (cmd :: ["-X"; "POST"] @ args)
```

Comment:
It's probably time we do the http requests through an OCaml library!
(unrelated to the nice work here, I know :) )

Reply:
agree! It's quite hackish, working but hackish

Reply:
yes, this is in the todolist with other libs (tar/git/etc.)